script/main.js

```diff
@@ -0,0 +1,181 @@
+}
+
+function editBook(bookIndex) {
+  $submitRecord.style.display = 'none';
+  $editRecord.style.display = 'block';
+  $formDiv.style.display = 'block';
+  $bookTitle.value = myLibraryArray[bookIndex].title;
+  $bookAuthor.value = myLibraryArray[bookIndex].author;
+  $bookPageNumber.value = myLibraryArray[bookIndex].pageNumber;
+
+  $editRecord = document.querySelector('#edit-record');
+  $editRecord.addEventListener('click', (e) => {
+    e.preventDefault();
+
+    const readStatus = getReadValue();
+    const editValue = {
+      title: $bookTitle.value,
+      author: $bookAuthor.value,
+      pageNumber: $bookPageNumber.value,
+      read: readStatus,
+    };
+
+    myLibraryArray.splice(bookIndex, 1, editValue);
+    updateLocalStorage();
+    render();
+    hideForm();
+  });
+}
+
+function changeStatus(book) {
+  if (myLibraryArray[book].read === 'Read') {
+    myLibraryArray[book].read = 'Not Read';
+  } else myLibraryArray[book].read = 'Read';
+}
+
+function findBook(libraryArray, name) {
+  let index = '';
+  if (libraryArray.length === 0 || libraryArray === null) {
+    return;
+  }
+
+  libraryArray.forEach((book) => {
+    if (book.title === name) {
+      index = libraryArray.indexOf(book);
+    }
+  });
+  return index;
+}
+
+function deleteBook(currentBookIndex) {
+  myLibraryArray.splice(currentBookIndex, 1);
+}
+
+$bookTable.addEventListener('click', (e) => {
+  const currentTarget = e.target.parentNode.parentNode.childNodes[1];
+  if (e.target.innerHTML === 'Delete') {
+    if (`are you sure you want to delete ${currentTarget.innerText}`) {
+      deleteBook(findBook(myLibraryArray, currentTarget.innerText));
+    }
+  }
+  if (e.target.innerHTML === 'Change Status') {
+    const bookIndex = findBook(myLibraryArray, currentTarget.innerText);
+    changeStatus(bookIndex);
+  }
+  if (e.target.innerHTML === 'Edit') {
+    const editIndex = findBook(myLibraryArray, currentTarget.innerText);
+    editBook(editIndex);
+  }
+  updateLocalStorage();
+  render();
+});
+
+function clearForm() {
+  $bookTitle.value = '';
+  $bookAuthor.value = '';
+  $bookPageNumber.value = '';
+}
+
+function addBook() {
+  if ($bookTitle.value.length === 0 || $bookAuthor.value.length === 0 || $bookPageNumber.value === '') {
+    errorMsg.style.color = 'red';
+    errorMsg.innerHTML = 'Please, fill all the fields';
+    return;
+  }
+  errorMsg.innerHTML = '';
+  const read = getReadValue();
+  const newBook = new Book($bookTitle.value, $bookAuthor.value, $bookPageNumber.value, read);
+  myLibraryArray.push(newBook);
+  updateLocalStorage();
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+  $newBook.addEventListener('click', () => {
+    $editRecord.style.display = 'none';
+    $formDiv.style.display = 'block';
+  });
+
+  document.querySelector('form').addEventListener('submit', (e) => {
+    e.preventDefault();
+    addBook();
+    render();
+    clearForm();
+    hideForm();
```

Comment:
- [ ] The form will always hide no matter how the validation goes. The user won't be able to see the  "Please, fill all the fields" message. I would suggest to create a function `isBookInputsValid()` which return true or false and if it returns false don't hide the form.

Reply:
isBookInputsValid() has been implemented